packages/provider-utils/src/schema.ts

```diff
@@ -62,7 +71,11 @@ function isSchema(value: unknown): value is Schema {
 }
 
 export function asSchema<OBJECT>(
-  schema: z.Schema<OBJECT, z.ZodTypeDef, any> | Schema<OBJECT> | undefined,
+  schema:
+    | z4.$ZodType<OBJECT, any>
+    | z3.Schema<OBJECT, z3.ZodTypeDef, any>
+    | Schema<OBJECT>
```

Comment:
wondering if it makes sense to extract this into a dedicated type e.g. extended schema